app/assets/javascripts/components/article_finder/article_finder.jsx

```diff
@@ -52,20 +55,38 @@ const ArticleFinder = createReactClass({
     }
   },
 
-  updateFields(key, value) {
-    return this.props.updateFields(key, value);
+  getQueryVariable() {
+    const query = window.location.search.substring(1);
+    const vars = query.split('&');
+    for (let i = 0; i < vars.length; i += 1) {
+        const pair = vars[i].split('=');
+        this.updateFields(pair[0], pair[1]);
+    }
+    },
```

Comment:
I think there might be a simpler way to solve this problem. It looks like this function is doing two jobs:
1. Parsing the query string
2. Updating each field

We already have the [query-string](https://www.npmjs.com/package/query-string) package installed, so I would suggest making use of it like so to solve this first problem:
```js
import qs from 'query-string';
//...
qs.parse(window.location.search) // { article_quality: "100", min_views: "0" ... }
```

Then you'll end up with an object to represent the query params. If you want to turn this into an array of arrays, you can do the following:
```js
const query = qs.parse(window.location.search)
Object.entries(query) // [ [ "article_quality", "100" ], [ "min_views", "0" ], ... ]
```

The reason you have `async/await` (which, is awesome!) is because we want all the updates to finish before we can build the URL, right? Typically when you have multiple asynchronous actions you would use `Promise.all`, which [evaluates an array of promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) and then returns an array with all of the final results. An async function will return a promise, so to create an array of promises you could do something like:
```js
const query = qs.parse(window.location.search);
const entries = Object.entries(query);
const updates = entries.map(([key, val]) => this.updateFields(key, val));
// At this point, the updates haven't been made yet!

Promise.all(updates).then(() => this.buildURL());
```

This reads a bit more cleanly to me. You can either remove or rename the `getQueryVariable` function

Reply:
The async await is there exactly for this reason , without async await it was not updating queries while calling buildURL() instantly , your use of query-string module is again great , sorry hadn't noticed the libraries installed already otherwise I would have used that !

Reply:
window.location.search and Object.entries(query) , both of them are producing a console warning. I'm a bit confused why did it happen?

Warning: Failed prop type: "value" must be a number or a range object

Reply:
I just found out that  the .updateFields(key,val) function is the reason for the warning . Any idea about how could I resolve this , 
`const updates = entries.map(([key, val]) => this.updateFields(key, val));`

Warning: Failed prop type: "value" must be a number or a range object

Reply:
Ah, I see now. So, I don't think the issue is any of the suggested changes but is rather the fact that when pulling values from the query string, each value will be a string when it might in fact need to be a number (e.g. in the case of `article_quality`).

Two options I can imagine here:
1. For the keys that we know will be a number, convert them to a number before calling `this.props.updateFields()`
1. Dynamically check if each value _could_ be a number. If it is, convert it, otherwise leave it as is

The second option seems more "clever" but I could see it introducing problems and might require a bit of commenting to make it understandable. I think either would work though.  :)

Reply:
I would add, I actually think that `.then(() => this.buildURL())` is probably unnecessary. I forgot you were building the URL on each change, so it should likely just work.

Reply:
I'm currently stuck on the dynamic part , everything else is done , would create a PR soon

Reply:
@hjhimanshu01 No worries! I think the first option will just fine. Good luck with exams!

Reply:
@bwreid , I just found out that to implement dynamic checking there is a small roadblock in the already implemented code , in the props that are passed to article_finder , "min_views" is stored as a string while "article_quality" as a number , I was stuck here , as I had been passing both as numbers which ideally should be the case as in dynamic checking as both are numbers.

The logic implemented is this :

`getParamsURL() {`
    `const query = qs.parse(window.location.search);`
    `const entries = Object.entries(query);`
   ` entries.map(([key, val]) => {`
     ` const copy_val = val ;`
      `val = (isNaN(parseInt(copy_val))) ? val : parseInt(val) ; `
     ` this.updateFields(key, val)`
   ` });`
   ` },`

Reply:
Update :

@bwreid @ragesoss  , encountered another bug that might have been there in the code for a while , details below :

On the Article Finder Page , if you click on the submit button without any search_term , it would fire up messages in console as well as a loading message that doesn't show anything.
![Screenshot from 2019-03-08 14-48-53](https://user-images.githubusercontent.com/22212482/54052166-b7ca0e00-41b1-11e9-83e6-f61e91483bed.png)

Reply:
Yes, it looks like that is the current behavior if you submit an empty query.

Feel free to add in a fix for that if it's very simple; alternatively, we could open a new issue for it.

Reply:
Let's have a new issue , I would take it up as this thread is going long , also I'm currently waiting for @bwreid for dynamic update checking issue where "article_quality" and "min_views" have a different data type , that is another issue if looked on or I could just keep it like the one which I had already made a commit with, making  a little changes for each one of the parameters.

Reply:
@hjhimanshu01 I would just go with the first option of only changing the value to a number for specific fields.